cds/webapi/C#/MetadataOperations/CDSWebApiServiceExtensions.cs

```diff
@@ -0,0 +1,2458 @@
+            oneToManyRelationship.Lookup = lookup;
+
+            try
+            {
+                uri = svc.PostCreate($"RelationshipDefinitions", JObject.FromObject(oneToManyRelationship));
+                string id = uri.ToString().Substring(uri.ToString().IndexOf("(") + 1, 36);
+                relationshipId = new Guid(id);
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+
+            try
+            {
+                LookupAttributeMetadata attribute = null;
+
+                var lookupLogicalName = lookup.LogicalName ?? lookup.SchemaName.ToLower();
+
+                attribute = svc.RetrieveAttribute<LookupAttributeMetadata>(
+                   oneToManyRelationship.ReferencingEntity,
+                   lookupLogicalName,
+                   "MetadataId");
+
+                attributeId = (Guid)attribute.MetadataId;
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+
+            if (!string.IsNullOrEmpty(solutionUniqueName))
+            {
+                try
+                {
+                    svc.AddSolutionComponent(relationshipId, SolutionComponentType.EntityRelationship, solutionUniqueName, true);
+                }
+                catch (Exception)
+                {
+                    throw;
+                }
+            }
+
+            return new CreateOneToManyResponse
+            {
+                RelationshipId = relationshipId,
+                AttributeId = attributeId
+            };
+        }
+
+        /// <summary>
+        /// Deletes an attribute
+        /// </summary>
+        /// <param name="svc">An instance of the CDSWebApiService to extend.</param>
+        /// <param name="entityLogicalName">The logical name of the entity that contains the attribute.</param>
+        /// <param name="attributeLogicalName">The logical name of the attribute to delete.</param>
+        public static void DeleteAttribute(this CDSWebApiService svc, string entityLogicalName, string attributeLogicalName)
+        {
+            try
+            {
+                svc.Delete("EntityDefinitions" +
+                    $"(LogicalName='{entityLogicalName}')/Attributes" +
+                    $"(LogicalName='{attributeLogicalName}')", STRONGCONSISTENCYHEADER);
+            }
+            catch (Exception)
+            {
+                throw new Exception($"There was a problem deleting the {attributeLogicalName} attribute in the {entityLogicalName} entity.");
+            }
+        }
+
+        /// <summary>
+        /// Deletes an entity by Id
+        /// </summary>
+        /// <param name="svc">An instance of the CDSWebApiService to extend.</param>
+        /// <param name="entityMetadataId">The MetadataId of the entity to delete.</param>
+        public static void DeleteEntity(this CDSWebApiService svc,
+            Guid entityMetadataId)
+        {
+            svc.DeleteEntity(entityMetadataId, null);
```

Comment:
For this method and the ones that follow, it is not recommended to have methods that do nothing other than call another method.

Reply:
There are two mutually exclusive ways to access many metadata items: by Metadata Id or by Name which is an alternate key.
See: [Retrieve metadata by name or MetadataId](https://docs.microsoft.com/en-us/powerapps/developer/common-data-service/webapi/retrieve-metadata-name-metadataid)

These methods do more than just call another method, they provide overrides that can be easily used depending on the developer's situation. One accepts a Guid id value, the other a string name value. Both of them call the same private method that can accept both and manage passing the correct one.

It is intentional and a means to simplify the developer experience while re-using the same underlying logic.